File: WdtcCore/src/main/java/org/wdt/wdtc/core/auth/accounts/Accounts.java
package org.wdt.wdtc.core.auth.accounts;

import org.wdt.wdtc.core.auth.User;
import org.wdt.wdtc.core.manger.FileManger;
import org.wdt.wdtc.core.manger.URLManger;
import org.wdt.wdtc.core.utils.StringUtils;

public class Accounts {
    private final AccountsType type;

    public Accounts() {
        this.type = User.getUser().getType() == AccountsType.Offline ? AccountsType.Offline : AccountsType.Yggdrasil;
    }

    public boolean ifAccountsIsOffline() {
        return type != AccountsType.Yggdrasil;
    }

    public String getJvmCommand() {
        return ifAccountsIsOffline()
                ? StringUtils.STRING_EMPTY
                : StringUtils.appendForString(StringUtils.STRING_SPACE, "-javaagent:", FileManger.getAuthlibInjector(),
                "=", URLManger.getLittleskinApi(), StringUtils.STRING_SPACE, "-Dauthlibinjector.yggdrasil.prefetched=", User.getUser().getAPIBase64());
    }


    public enum AccountsType {
        Offline,
        Yggdrasil
    }

}

File: WdtcCore/src/main/kotlin/org/wdt/wdtc/core/auth/accounts/Accounts.kt
package org.wdt.wdtc.core.auth.accounts

import org.wdt.wdtc.core.auth.preferredUser
import org.wdt.wdtc.core.manger.authlibInjector
import org.wdt.wdtc.core.manger.littleskinApiUrl
import org.wdt.wdtc.core.utils.STRING_EMPTY
import org.wdt.wdtc.core.utils.STRING_SPACE

object Accounts {
  private val type: AccountsType
  private val user = preferredUser

  init {
    type = user.type
  }

  private val ifAccountsIsOffline = type != AccountsType.YGGDRASIL

  val jvmCommand: String
    get() = if (ifAccountsIsOffline)
      STRING_EMPTY
    else buildString {
      append(STRING_SPACE)
      append("-javaagent:")
      append(authlibInjector.canonicalPath)
      append("=")
      append(littleskinApiUrl)
      append(STRING_SPACE)
      append("-Dauthlibinjector.yggdrasil.prefetched=")
      append(user.base64Data)
    }

  enum class AccountsType {
    OFFLINE,
    YGGDRASIL
  }
}
